src/main/java/duke/interact/Parser.java

```diff
@@ -0,0 +1,142 @@
+package duke.interact;
+
+import duke.TaskList;
+import duke.command.Command;
+import duke.command.CommandAdd;
+import duke.command.CommandBye;
+import duke.command.CommandDel;
+import duke.command.CommandDone;
+import duke.command.CommandFind;
+import duke.command.CommandList;
+import duke.command.CommandUpDate;
+import duke.command.CommandUpDetail;
+import duke.exception.DukeException;
+import duke.exception.DukeExceptionCommand;
+import duke.exception.DukeExceptionDate;
+import duke.exception.DukeExceptionDateFormat;
+import duke.exception.DukeExceptionDescription;
+import duke.exception.DukeExceptionIndex;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parses user input and checks whether the user has given wrongly formatted input.
+ */
+public class Parser {
+
+    /**
+     * Returns a Command representing the user input.
+     * @param line String input by user.
+     * @param tasks Current TaskList.
+     * @return Command representing user input.
+     * @throws DukeException Exception thrown if user input is wrongly formatted.
+     */
+    public Command parse(String line, TaskList tasks) throws DukeException {
+        int size = tasks.getSize();
+        switch (line) {
+        case "bye":
+            return new CommandBye();
+
+        case "list":
+            return new CommandList();
+
+        default:
+            String[] comArs = line.split("\\s", 2);
+            String inputCommand = comArs[0];
+
+            int index;
+            String details;
+            String[] msgAndDate;
+            String[] indexAndNew;
+
+            switch (inputCommand) {
+            case "done":
+                if (comArs.length == 1) {
+                    throw new DukeExceptionIndex("done");
+                }
+                index = Integer.parseInt(comArs[1]) - 1;
```

Comment:
Method is a little too long. Perhaps can shift some of the method body to separate private methods.